src/transformers/data/data_collator.py

```diff
@@ -305,20 +305,30 @@ def torch_call(self, features):
 
         label_name = "label" if "label" in features[0].keys() else "labels"
         labels = [feature[label_name] for feature in features] if label_name in features[0].keys() else None
+
+        if labels is not None:
+            no_label_features = [{k: v for k, v in feature.items() if k != label_name} for feature in features]
+        else:
+            no_label_features = features
+
         batch = self.tokenizer.pad(
-            features,
+            no_label_features,
             padding=self.padding,
             max_length=self.max_length,
             pad_to_multiple_of=self.pad_to_multiple_of,
-            # Conversion to tensors will fail if we have labels as they are not of the same length yet.
             return_tensors="pt" if labels is None else None,
```

Comment:
```suggestion
            return_tensors="pt",
```
Can always be this now.